lib/src/realm_object.dart

```diff
@@ -366,3 +367,8 @@ class RealmObjectNotificationsController<T extends RealmObject> extends Notifica
     streamController.addError(error);
   }
 }
+
+/// @nodoc
+class DynamicRealmObject with RealmEntity, RealmObject {
```

Comment:
I think we should overwrite `noSuchMethod` here to allow reading/writing any property on this object. Much better than using `RealmObject.get<XYZ>();`

Reply:
I see there is a previous discussion about this. I think @nielsenko is right and we could support dynamic with `noSuchMethod`. At the same time this object will have the ability to work with `.get<int>()` cause its part of the parent API

Reply:
This is covered in a different PR - https://github.com/realm/realm-dart/pull/524. The current PR only exposes dynamic API on the Realm class, not the RealmObject. The introduction of `DynamicRealmObject` is necessary for some primitive testing, but I decided to split the two tasks as each is fairly large and hoped it will make reviews easier.

Reply:
ok I see